storage.go

```diff
@@ -124,9 +124,19 @@ func (s *Storage) stat(ctx context.Context, path string, opt pairStorageStat) (o
 	rp := s.getAbsPath(path)
 	fl, err := s.connection.List(rp)
 	if err != nil {
-		return nil, err
+		flst, err := s.connection.List(filepath.Dir(rp))
+		if err != nil {
+			return nil, err
+		}
+		for i := range flst {
```

Comment:
Can you provide more information about this?

- Which error message could be returned if we try to list a file?


According the code from `fclairamb/ftpserverlib`, it supports stat file:

https://github.com/fclairamb/ftpserverlib/blob/53cb37b53a67afc0c3cff1adada56b35568275f6/handle_files.go#L416-L450

Reply:
BTW, I found `ftpserverlib` support stat a file:  https://sourcegraph.com/github.com/fclairamb/ftpserverlib@53cb37b53a67afc0c3cff1adada56b35568275f6/-/blob/handle_misc.go?L54:35

And the behavior is allowed in spec:

```txt
STATUS (STAT)

            This command shall cause a status response to be sent over
            the control connection in the form of a reply.  The command
            may be sent during a file transfer (along with the Telnet IP
            and Synch signals--see the Section on FTP Commands) in which
            case the server will respond with the status of the
            operation in progress, or it may be sent between file
            transfers.  In the latter case, the command may have an
            argument field.  If the argument is a pathname, the command
            is analogous to the "list" command except that data shall be
            transferred over the control connection.  If a partial
            pathname is given, the server may respond with a list of
            file names or attributes associated with that specification.
            If no argument is given, the server should return general
            status information about the server FTP process.  This
            should include current values of all transfer parameters and
            the status of connections.
```